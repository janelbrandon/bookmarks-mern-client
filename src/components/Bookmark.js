import React from 'react'
import {removeBookmark} from '../services/BookmarkService'

function Bookmark (props) {
  const { _id, title, url, remove } = props

  return (
    <li>
      {title} (<a href={url} target="_blank">Visit</a>)
      <button onClick={ () => remove ? remove(_id) : removeBookmark(_id) }>Delete!</button>
    </li>
  )
}

export default Bookmark
