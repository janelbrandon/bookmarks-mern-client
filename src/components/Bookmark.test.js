import React from 'react'
import Bookmark from './Bookmark'
import renderer from 'react-test-renderer'

describe('Bookmark', () => {
  it('should render as expected', () => {
    const tree = renderer.create(
      <Bookmark _id='123' title='My Bookmark' url='http://mybookmark.com' remove={ () => {}}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
