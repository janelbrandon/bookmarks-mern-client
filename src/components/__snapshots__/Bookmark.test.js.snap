// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark should render as expected 1`] = `
<li>
  My Bookmark
   (
  <a
    href="http://mybookmark.com"
    target="_blank"
  >
    Visit
  </a>
  )
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
